implementation/jobberknoll/api/shared/openapi.test.ts

```diff
@@ -0,0 +1,71 @@
+import { z } from "@hono/zod-openapi";
+import { uuid } from "@jobberknoll/core/shared";
+import { assert, assertEquals } from "@std/assert";
+import { errorDto, IdParamSchema, jsonRes } from "./openapi.ts";
+
+Deno.test("jsonRes should return a response object", () => {
+  const TestSchema = z.literal("test");
+  const description = "A test schema.";
+
+  const result = jsonRes(TestSchema, description);
+
+  assertEquals(result, {
+    content: { "application/json": { schema: TestSchema } },
+    description,
+  });
+});
+
+Deno.test("errorDto should accept correct errors", () => {
+  const TestDto = errorDto("TestDto", 400, "test", "A test DTO.");
+  const testError = {
+    code: 400,
+    kind: "test",
+    messageEn: "Test error message.",
+  };
+
+  const result = TestDto.safeParse(testError);
+
+  assert(result.success);
+});
+
+Deno.test("errorDto should reject mismatched codes", () => {
+  const TestDto = errorDto("TestDto", 400, "test", "A test DTO.");
+  const testError = {
+    code: 401,
+    kind: "test",
+    messageEn: "Test error message.",
+  };
+
+  const result = TestDto.safeParse(testError);
+
+  assert(!result.success);
+});
+
+Deno.test("errorDto should reject mismatched kinds", () => {
+  const TestDto = errorDto("TestDto", 400, "test", "A test DTO.");
+  const testError = {
+    code: 400,
+    kind: "test2",
+    messageEn: "Test error message.",
+  };
+
+  const result = TestDto.safeParse(testError);
+
+  assert(!result.success);
+});
```

Comment:
How about doing some parameterized testing here?

Reply:
If this is even possible in your framework xd

Reply:
I'm not even using a testing framework, `Deno.test` is a built-in testing primitive. I could import a whole framework but honestly I don't feel the need. I can make it parametrized by just wrapping in a good old `for` loop (like in `/tests/docs-e2e.test.ts`).

Will do 👍